v0/destinations/engage/utils.js

```diff
@@ -0,0 +1,104 @@
+// one for lists
+const { set } = require("lodash");
+const { getDestinationExternalID } = require("../../util");
+
+/**
+ * @param {*} message
+ * @param {*} Config
+ * @returns the listIds from the payload based on priority and availability
+ */
+const getLists = (message, Config) => {
+  let listIds = getDestinationExternalID(message, "engageListId");
+  if (!listIds) {
+    listIds = [];
+    const objectlist = Config?.listIds;
+    if (objectlist) {
+      objectlist.forEach(v => {
+        listIds = listIds.concat(Object.values(v));
+      });
+    }
+  }
+  return listIds;
+};
+
+/**
+ * @param {*} message
+ * @returns the Engage User ID based on priority and availability
+ */
+const getUID = message => {
+  let engageId = getDestinationExternalID(message, "engageId");
+  if (!engageId) {
+    const { context, traits, anonymousId, userId } = message;
+    if (
+      userId ||
+      traits?.userId ||
+      traits?.id ||
+      context?.traits?.userId ||
+      context?.traits?.id ||
+      anonymousId
+    )
+      engageId =
+        userId ||
+        traits.userId ||
+        traits.id ||
+        context.traits.userId ||
+        context.traits.id ||
+        anonymousId;
+  }
```

Comment:
if (getFieldValueFromMessage(message, "userId")) {
    engageId = getFieldValueFromMessage(message, "userId");
}

Reply:
Should we leave the `anonymousId` or check for it if userId is not present ?

Reply:
yes, lets write `userIdOnly` then

Reply:
Yeah we don't have